_posts/2023-06-06-Client-cert-auth-without-roles.adoc

```diff
@@ -0,0 +1,252 @@
+
+Next, convert the client keystores into PKCS12 format and import them into your browser so you can pick which one to 
+present to the server later on:
+
+```
+keytool -importkeystore -srckeystore client1.keystore -srcstoretype jks -destkeystore client1.keystore.pkcs12 -deststoretype pkcs12 -srcstorepass keystorepass -deststorepass keystorepass
+
+keytool -importkeystore -srckeystore client2.keystore -srcstoretype jks -destkeystore client2.keystore.pkcs12 -deststoretype pkcs12 -srcstorepass keystorepass -deststorepass keystorepass
+
+```
+
+Finally, copy the server.keystore and server.truststore files to your WildFly server instance:
+
+```
+
+cp $PATH_TO_ELYTRON_EXAMPLES/dynamic-certificates/target/server.* $WILDFLY_HOME/standalone/configuration
+
+```
+
+== Server Configuration
+
+The following set of instructions are going to be used to configure the wildfly server in order to 
+establish the mutual TLS connection. We will be deploying a simple web application under the 
+`elytron-examples/client-cert-auth-without-roles/simple-webapp` folder.
+
+Please navigate to the elytron server home directory and enter the following command.
+
+```
+
+$SERVER_HOME/bin/jboss-cli.sh --connect 
+
+```
+
+This will connect you to the server, after which you may proceed to configuring the server to establish the connection. 
+
+First we need to create two keystores inside the elytron subsystem. This will tell elytron where to go inder to find the 
+keystores. The command for this is: 
+
+```
+/subsystem=elytron/key-store=twoWayKS:add(path=/path/to/server.keystore, relative-to=jboss.server.config.dir, credential-reference={clear-text=keystorepass}, type=JKS)
+/subsystem=elytron/key-store=twoWayTS:add(path=/path/to/server.truststore,relative-to=jboss.server.config.dir, credential-reference={clear-text=truststorepass}, type=JKS)
+```
+
+The `relative-to` option specifies that the path that you have entered is relative to the `/standalone/configuration` folder 
+inside your wildfly repository. 
+
+Next, we need to create a key-manager and a trust manager to point to `twoWayKS` and `twoWayTS`, the previously 
+created keystores using the following commands: 
+```
+/subsystem=elytron/key-manager=twoWayKM:add(key-store=twoWayKS,credential-reference={clear-text=keystorepass})
+
+/subsystem=elytron/trust-manager=twoWayTM:add(key-store=twoWayTS)
+```
+
+We also need to have an identity realm, which would help up map our user to one that exists inside the file system. 
+For this step, let us create a filesystem realm called `idsRealm`. We will enter into this realm our identity with 
+name `client-cert-user` which we will map the certificate into. 
+
+```
+/subsystem=elytron/filesystem-realm=idsRealm:add(path=ids, relative-to=jboss.server.config.dir)
+/subsystem=elytron/filesystem-realm=idsRealm:add-identity(identity=client-cert-user)
+```
+
+Next, we need to configure our server-ssl-context. This is a collection of ciphers, TLS settings and certificates, like the 
```

Comment:
`server-ssl-context`

Reply:
I think we can remove " This is a collection of ciphers, TLS settings and certificates, like the ones we will be using to authenticate our server and the client."